integration_tests/tests/new_account.rs

```diff
@@ -62,3 +62,71 @@ fn test_new_account_duplicate() {
         )))
     );
 }
+
+fn new_account_case_insensitive_username() -> Result<(), TestError> {
+    let account = generate_account();
+
+    client::new_account::send(
+        api_loc(),
+        &NewAccountRequest {
+            username: account.username.clone(),
+            auth: AuthServiceImpl::<ClockImpl, RsaImpl>::generate_auth(&account).unwrap(),
+            public_key: serde_json::to_string(&account.keys.to_public_key()).unwrap(),
+        },
+    )?;
+
+    client::new_account::send(
+        api_loc(),
+        &NewAccountRequest {
+            username: account.username.to_uppercase(),
+            auth: AuthServiceImpl::<ClockImpl, RsaImpl>::generate_auth(&account).unwrap(),
+            public_key: serde_json::to_string(&account.keys.to_public_key()).unwrap(),
+        },
+    )?;
+
+    Ok(())
+}
+
+#[test]
+fn test_new_account_case_insensitive_username() {
+    assert_matches!(
+        new_account_case_insensitive_username(),
+        Err(TestError::NewAccountError(new_account::Error::API(
+            NewAccountError::UsernameTaken
+        )))
+    );
+}
+
+fn new_account_alphanumeric_username() -> Result<(), TestError> {
+    let account = generate_account();
+
+    client::new_account::send(
```

Comment:
This test doesn't need to make two accounts, I was just being clever with the last one. The idea was that if I made two accounts that were the same the second one would give me a `NewAccountError::UsernameTaken`, so I made one with lowercase and one with uppercase and checked for that error